### PR TITLE
Remove the java.debug option from commands and introduce an env. variable to enable debugging

### LINKED ISSUE #10328: Remove the "java.debug" option/flag from Ballerina Commands
**Description:**
Currently Java debug is enabled by including the "--java.debug" flag with a command such as Ballerina Run.
```cmd
ballerina run --java.debug 5005 test.bal
```

Java debugging is enabled via the Ballerina start up script, and the flag would be picked up wherever specified. However, "java.debug" is also specified as an option to Ballerina commands in order to ignore the "--java.debug" argument and the port argument specified after it from the actual argument list.

With https://github.com/ballerina-platform/ballerina-lang/pull/9991, this would not work with the run command if "--java.debug" is specified after the source to run.

As discussed in https://groups.google.com/d/topic/ballerina-dev/Pceq8lTVCpU/discussion, the "java.debug" option/flag needs to be removed from the Ballerina commands.

**Possible alternatives are:**
1. Reconstruct the argument list removing "--java.debug" and the argument after it
2. Enable Java debugging via an environment variable

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
This PR removes the `java.debug` option from Ballerina commands, and introduce an environment variable `BAL_JAVA_DEBUG` which could be set to enable debugging.

Fixes #10328

### CODE REVIEW

**Comment by MaryamZi** [comment]
Not to be merged yet, needs to be tested more on different OSs.

File: distribution/zip/ballerina/bin/ballerina [hunk 1]
```diff
@@ -136,25 +136,17 @@ fi
 
 # ----- Process the input command ----------------------------------------------
 
-for c in "$@"
-do
-    if [ "$c" = "--java.debug" ] || [ "$c" = "-java.debug" ] || [ "$c" = "java.debug" ]; then
-          CMD="--java.debug"
-    elif [ "$CMD" = "--java.debug" ] && [ -z "$PORT" ]; then
-          PORT=$c
+if [ -n "${BAL_JAVA_DEBUG+set}" ]; then
```
Comment (Shan1024) [inline distribution/zip/ballerina/bin/ballerina]: Why do we need `+set`?

Comment (MaryamZi) [reply distribution/zip/ballerina/bin/ballerina]: This is to distinguish between `BAL_JAVA_DEBUG` not being set and `BAL_JAVA_DEBUG` being set to an empty string/nothing [1].

For example if it is set to an empty string we want to say "Please specify the debug port for the BAL_JAVA_DEBUG variable"

[1] https://serverfault.com/a/382740

**Comment by MaryamZi** [comment]
Sample usage on Linux, avoiding exporting:
```cmd
$ BAL_JAVA_DEBUG=5005 ballerina run test.bal
```

**Pull request merged**